### PR TITLE
feat(puppeteer): add device descriptors

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/puppeteer/blob/master/DeviceDescriptors.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

I am looking for input here, we already have existing definitions of the types required here but they do have different nullability, any suggestions?

### CODE REVIEW

File: types/puppeteer/DeviceDescriptors.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  interface Device {
+    name: string;
+    userAgent: string;
+    viewport: Viewport;
+  }
+}
+
+declare const devices: {
+  [key: string]: devices.Device;
```
[the remark this answers is not in the document]
Comment (SimonSchick) [reply types/puppeteer/DeviceDescriptors.d.ts]: Can you elaborate?

[the remark this answers is not in the document]
Comment (amcasey) [reply types/puppeteer/DeviceDescriptors.d.ts]: Sorry, that was a little vague.  As written, in addition to being a type, `devices.Viewport` will be an expression of type `devices.Device`.  That's probably not intended, but there isn't a good way to express that the indexer doesn't cover the names "Viewport" and "Device".

**Comment by SimonSchick** [comment]
@amcasey I was kinda hoping to get some feedback on my comment regarding duplication ¯\_(ツ)_/¯.

**Comment by amcasey** [comment]
@SimonSchick sorry, which comment?

**Comment by SimonSchick** [comment]
>I am looking for input here, we already have existing definitions of the types required here but they do have different nullability, any suggestions?

**Comment by amcasey** [comment]
Ah.  I can't speak for all the maintainers but, personally, I find it easy to miss comments after a block of boilerplate.  You might have better luck posting them as separate, top-level comments.

As to your specific question, I think I need more context.  Do you have a link to the other types?

**Comment by SimonSchick** [comment]
See https://github.com/SimonSchick/DefinitelyTyped/blob/7b8c006a78eed82402b8ba6b8decb88736aa5ef6/types/puppeteer/index.d.ts#L1971
and
https://github.com/SimonSchick/DefinitelyTyped/blob/7b8c006a78eed82402b8ba6b8decb88736aa5ef6/types/puppeteer/index.d.ts#L443

**Comment by amcasey** [comment]
So your question is how to capture the fact that a property is optional in some contexts and not in others?

**Pull request merged**